datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -1613,7 +1757,10 @@ private void scheduleExpiryTimerAfterSessionTimeout() {
   @VisibleForTesting
   void onSessionExpired() {
     LOG.error("Zookeeper session expired.");
+    //cancel the lock clean up
+    _orphanLockCleanupFuture.cancel(true);
     onBecomeFollower();
+    connect();
```

Comment:
This is a temporary change. This is to make sure that we keep the execution path same till we dont fix the connect/disconnect. What will happen is after session expiry, it will try to disconnect and the zkThread will die and the instance will not try to join the cluster. This will be removed when we are ready with handling new session. Most of the things are already covered. Handling new session should be a small change, but I want to keep it separate.

Reply:
Got it. In your other PR to bring tasks down, I noticed you had a comment explaining that this is temporary. Want to add it here? Or add this as an action item in some document where you're tracking all the changes needed. Don't want to lose sight of these things.